### PR TITLE
Fixing a long-standing bug with Richardson iteration

### PR DESCRIPTION
**Pull request by wdhawkins**
Richardson iteration in PETSc is a Krylov method, and we were failing to perform a final sweep with converged phi and delayed psi to recover the true solution.  This PR fixes that oversight and changes most internal and user-facing references to `richardson` to `krylov_richardson` to distinguish PETSc Richardson from classic Richardson.

### CODE REVIEW

File: framework/math/linear_solver/linear_solver.cc [hunk 1]
```diff
@@ -82,7 +82,14 @@ LinearSolver::Setup()
...
   KSPCreate(opensn::mpi_comm, &ksp_);
-  KSPSetType(ksp_, iterative_method_.c_str());
+
+  // In OpenSn the PETSc version of Richardson iteration is referred to as krylov_richardson to
+  // distinguish it from classic Richardson. At this point, we need to convert from
+  // krylov_richardson to the correct PETSc algorithm name.
+  if (iterative_method_ == "krylov_richardson")
+    KSPSetType(ksp_, "richardson");
+  else
+    KSPSetType(ksp_, iterative_method_.c_str());
```
Comment (ragusa) [inline framework/math/linear_solver/linear_solver.cc:92]: that's the part I was missing !!

**Review by ragusa — APPROVED** [review]

File: modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/lbs_iterative_methods.h [hunk 2]
```diff
@@ -12,8 +12,8 @@ namespace lbs
 enum class IterativeMethod : int
 {
   NONE = 0,
-  CLASSICRICHARDSON = 1,        ///< Otherwise known as Source Iteration
-  CLASSICRICHARDSON_CYCLES = 2, ///< Source Iteration with Cycles support
+  // CLASSICRICHARDSON = 1,        ///< Otherwise known as Source Iteration
+  // CLASSICRICHARDSON_CYCLES = 2, ///< Source Iteration with Cycles support
```
Comment (andrsd) [inline modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/lbs_iterative_methods.h:16]: Are there any plans on recycling those 2 values in future? If not, then delete those lines...

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/lbs_iterative_methods.h (hunk 2 above)
Comment (wdhawkins) [reply modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/lbs_iterative_methods.h:16]: Yes, classic Richardson will make a comeback as soon as possible. I believe it's even slated for 1.0.

Comment (andrsd) [reply modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/lbs_iterative_methods.h:16]: Ok.

**Review by andrsd — APPROVED** [review]

**Pull request merged**